model_analyzer/record/metrics_mapper.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .gpu_free_memory import GPUFreeMemory
+from .gpu_used_memory import GPUUsedMemory
+from .gpu_utilization import GPUUtilization
+from .cpu_used_memory import CPUUsedMemory
+from .cpu_free_memory import CPUFreeMemory
+from .perf_throughput import PerfThroughput
+from .perf_latency import PerfLatency
+
+
+class MetricsMapper:
+    """
+    Class maps metrics to record types
+    """
+
+    tag_to_metric = {
```

Comment:
We can remove the need for `tag_to_metric` array using a code like below:

```py
import abc
import importlib


class BaseProvider(abc.ABCMeta):
    providers = {}

    def __new__(cls, name, bases, namespace):
        instance = abc.ABCMeta.__new__(cls, name, bases, namespace)

        if isinstance(instance.name, str):
            cls.providers[instance.name] = instance

        return instance

    @classmethod
    def get(cls, name):
        if name not in cls.providers:
            try:
                importlib.import_module('model_analyzer.record.%s' % name)
            except ImportError as e:
                print(e)
        return cls.providers[name]


class Provider(metaclass=BaseProvider):
    def __init__(self):
        pass

    @property
    @abc.abstractmethod
    def name(self):
        raise NotImplemented()
```

This assumes that we have a name field for each of the records.

Reply:
The name field being equal to the filename right? This seems to require that the class is hardcoded with the name of the file/module.

Reply:
I'm not sure how we would modify the metrics we measure using this method either?

Reply:
Yes exactly, we just need to add a name to the field to the subclasses that is equal to the file name. I can do this too if it looks good to you.

Reply:
This means that the config options for the metrics will have to be translated into the filenames as well, or we would have to use the filenames themselves to configure which metrics are being measured.

Reply:
Yes, we would have to use the filenames. I think our filenames look good but it can be subjective 😆

Reply:
Done

Reply:
Looks great!